<a href="https://colab.research.google.com/github/Allajka/statistics/blob/main/seminar_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Задание 1

Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

Доверительный интервал для средних арифметических. Сигма совокупности известна:

$$\overline{x} ± z_\frac{a}{2} * \frac{σ}{\sqrt{n}}$$
$$\overline{x} = 80, σ = 16, n = 256$$
$$z_\frac{a}{2}(a = 0.05/2) = 1,96 $$

$$80 ± (1,96 * \frac{16}{\sqrt{256}})$$

In [1]:
80 + (1.96 * 16/256**0.5)

81.96

In [2]:
80 - (1.96 * 16/256**0.5)

78.04

**Ответ: 95%-й доверительный интервал для оценки матиматического ожидания генеральной совокупности: [78,04 ; 81,96]**

# Задание 2

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1

Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

Доверительный интервал для средних арифметических. Сигма совокупности неизвестна: 
$$\overline{x} ± t_\frac{a}{2} * \frac{σ}{\sqrt{n}}$$

In [3]:
import numpy as np
import scipy.stats as stats

In [5]:
values_X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]) 
mean_values_X = round(np.mean(values_X), 3) # Среднее арифметиическое
mean_values_X

6.59

In [6]:
dispersion = round(np.var(values_X, ddof=1), 2) # несмещенная дисперсия
dispersion

0.2

In [13]:
t_tabular = stats.t.ppf(0.975, 9) # t_табличное, 100-5/2 и 10-1
t_tabular

2.2621571627409915

In [14]:
mean_values_X + t_tabular * np.sqrt(dispersion/len(values_X))

6.909917333976775

In [15]:
mean_values_X - t_tabular * np.sqrt(dispersion/len(values_X))

6.2700826660232245

**Ответ: 95%-й доверительный интервал для истинного значения величины X : [6.270 ; 6.910]**

# Задание 3

Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170

Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175

Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

Интервальная оценка для разности средних арифметических:

$$Δ ± t_\frac{a}{2} * S_Δ$$
где,
$$Δ(дельта) = \overline{X}_1 - \overline{X}_2$$

Выборки могут иметь разную дисперсию, поэтому нужно найти среднюю: 
$$D = \frac{1}{2}(D_1+D_2)$$
$$S_Δ = \sqrt{\frac{D}{n_1}+\frac{D}{n_2}} $$

In [16]:
import numpy as np
import scipy.stats as stats

In [17]:
mothers = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175]) 
daughters = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170]) 

In [18]:
# Среднее арифметиическое
mean_mothers = np.mean(mothers) 
mean_daughters = np.mean(daughters) 
print(f'Среднее арифметиическое рост матерей - {mean_mothers}, рост дочерей {mean_daughters}')

Среднее арифметиическое рост матерей - 168.1, рост дочерей 166.2


In [20]:
delta = round(mean_mothers - mean_daughters, 3)
delta

1.9

In [22]:
# несмещенная дисперсия
D1 = round(np.var(mothers, ddof=1), 3) 
D2 = round(np.var(daughters, ddof=1), 3) 
print(f'Несмещенная дисперсия рост матерей - {D1}, рост дочерей {D1}')

Несмещенная дисперсия рост матерей - 60.1, рост дочерей 60.1


In [23]:
# объединенная оценка дисперсий 
D = (D1 + D2) / 2 
D

75.5835

In [24]:
SE = np.sqrt(D/len(mothers) + D/len(daughters))  # стандартная ошибка разности средних 
SE

3.888020061676637

In [25]:
t_tabul = stats.t.ppf(0.975, len(mothers) + len(daughters) - 2) # t_табличное, 100-5/2 и (10-1)*2 = 18
t_tabul

2.10092204024096

In [26]:
L = delta - t_tabul * SE
U = delta + t_tabul * SE
print(f"Доверительный интервал [{L}; {U}]")

Доверительный интервал [-6.268427040475464; 10.068427040475465]


**Ответ:  95% доверительный интервал для разности среднего роста родителей и детей [-6.268; 10.068]**